Documentation/writing-tizen-native-apps.md

```diff
@@ -0,0 +1,88 @@
+# Writing a Tizen Native App
+
+This document guides how to write a Tizen Native (C/C++) app with nnstreamer APIs (Tizen Machine-Learning Inference API Set).
+
+## About the Tizen ML-Inference API Sets
+
+There are two API sets as of Tizen 5.5 M2, Machine-Learning Inference: "Pipeline" and "Single".
+
+With **Pipeline** API set, you can construct a GStreamer pipeline with nnstreamer elements and gstreamer elements in the "whitelist". GStreamer plugins not listed in the "whitelist" is not supported by Pipelie APIs. Note that an alias element, ```tizencamvideosrc``` (or ML_TIZEN_CAM_VIDEO_SRC macro), denotes for a video camera source, which may vary per device; in other words, you don't need to worry whether it's ```v4l2src``` or ```camsrc``` or whatsoever. For the list of whitelist, refer to the Tizen API doc or ```/etc/nnstreamer.ini```.
+
+With **Single** API set, you can invoke a neural network model with a input tensor/tensors without constructing a full pipeline.
+
+## General Flow / Single
+
+1. Get a handle (open a model)
+
+2. (Optional) Investigate the required input/output types (dimensions) with utility functions.
+
+3. Invoke (provide an input, get an output)
+
+4. Close the handle
+
+Note that set_timeout API allows you to set timeout for invoke function.
+
+## General Flow / Pipeline
+
+It is far more rich featured and provides a lot more functionalities. However, the developer is required to understand GStreamer pipelines. For GStreamer pipelines in general, plrease refer to GStreamer documents.
+
+1. Construct a pipeline (ml\_pipeline\_construct) and get a pipeline handle.
+
+2. (Optional) Attach callbacks or get other handles from the constructed pipeline. In order to attach callbacks or get handles, the corresponding elements should have names defined so that they can be recognized in run-time.
+
+3. Start the pipeline. The callbacks will be invoked during the pipeline execution. You may control it with the additional handles (e.g., input/output switches, valves).
+
+4. Pause/Resume the pipeline.
```

Comment:
we dont have APIs to directly access the element to pause/resume (stop/start only)

Reply:
Fixed.